packages/insea-connect-backend/src/main/java/ma/insea/connect/chat/group/GroupService.java

```diff
@@ -68,9 +77,13 @@ public List<User> findUsers(Long groupId) {
 
     }
     public ResponseEntity<String> addGroupMembers(Long groupId, List<Long> users) {
-        //test if the connected user is an admin
-        
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        User connectedUser = userRepository.findByUsername(authentication.getName()).orElse(null);
+        Membership membership = membershipRepository.findByUserIdAndGroupId(connectedUser.getId(), groupId);
         
+        if(membership == null || !membership.getIsAdmin()) {
+            return ResponseEntity.badRequest().body("You are not allowed to add members to this group");
```

Comment:
the status code in this case should be 401 unauthorized not 400 bad request.
Another point is that Service should not handle Response entities, That's the responsibility of a controller, if you want to raise an exception, you can simply create a custom exception and make it produce a 401 status code, then you can just throw it in the service with a custom message

Reply:
done